valkyrie/lib/valkyrie/persistence/postgres/queries/find_by_id_query.rb

```diff
@@ -15,7 +15,10 @@ def run
     private
 
       def relation
-        orm_model.find(id)
+        resource = orm_model.find_by(secondary_identifier: id)
+        raise ActiveRecord::RecordNotFound unless resource
+
+        resource
```

Comment:
This seems to only allow lookup by secondary/external ID. Is that right? Intended?